alkos/kernel/CMakeLists.txt

```diff
@@ -10,6 +10,9 @@ include_directories(${SYSROOT}/usr/include)
 # Define ARCH_DIR
 set(ARCH_DIR "${CMAKE_CURRENT_SOURCE_DIR}/arch/${ARCH}")
 
+# Additional test definitions
+# add_compile_definitions(ALKOS_TEST=0)
```

Comment:
Why is it commented out? If some important switch in this program requires commenting in & out this code - then something is probably wrong.

Reply:
This test basically kills the kernel, cause it checks if panic is invoked when stack smashing occurs. Means that you should provide the flag directly to cmake or uncomment this code which does it for you

Reply:
Well then it's bad design? Will we have merge conflicts now because someone uncomments / comments this line to do some testing?

Reply:
What merge conflicts? This should always be commented, and test builds should only be performed when necessary. Leaving a compile define enabled for testing—which essentially destroys the kernel at the entry point—is a clear sign of either a complete lack of awareness or outright negligence.

Reply:
What we can conclude from that we need to design solid test framework as mentioned many times.